### ISSUE TITLE
Training on datasets with only ONE bounding box

### ISSUE THREAD

**Issue by SebastianLeborg** [pierluigiferrari/ssd_keras#98]
Hello!

I'm building a object detection model for garment identification. I'm trying to identify top, bottom or whole-body garments given any image of a clothed person. My current dataset is DeepFashion (http://mmlab.ie.cuhk.edu.hk/projects/DeepFashion.html) which is quite good but has a pretty big problem in that it only provides one bounding box per image. This bounding box can be either the top (shirt/sweater/etc), bottom (pants) or a whole-body garment (dress). 

In the case that the image contains both a top and a bottom garment, the label simply specifies one of the classes and the bounding box of that one class. This means that when I train the network I don't have the full list of correct labels which means that the loss function will consider a correctly predicted bounding box of a class that is not specified in the label as an incorrect prediction. When trained for quite some time, the network seems to converge towards outputting 'whole-body garment' or nothing at all instead of creating two different bounding boxes for top and bottom.

I know this is more of a problem with the dataset than this specific implementation but I'm interested in if someone has any intuition on how to tackle this problem. One idea I had was to reduce the loss to 0 for any predictions that does not correspond to the class of the label. Does anyone know how to do this or have a better idea?

**Comment by pierluigiferrari** [comment]
Among several bad alternatives, the idea you mentioned of reducing the loss for non-label predictions to 0 seems like the least bad alternative. Another alternative would be to take a small subset of the Deep Fashion dataset (a few thousand images), annotate that subset properly for object detection, and then train only on that with heavy data augmentation. Not ideal either of course.

As you mentioned above, training the network on the dataset as is won't work, because you would feed the model contradictory ground truth data. If a bottom is not annotated in an image, the model would be punished for predicting it as a bottom, and if a bottom is annotated in an image, the model would be punished for predicting it as background.
